plugins/kn-migration/cmd/client.go

```diff
@@ -0,0 +1,235 @@
+	revision.ObjectMeta.Namespace = cl.namespace
+	revision.ObjectMeta.ResourceVersion = ""
+	revision.ObjectMeta.OwnerReferences[0].UID= config_uuid
+	revision.ObjectMeta.Labels["serving.knative.dev/configurationGeneration"] = originalrevision.ObjectMeta.Labels["serving.knative.dev/configurationGeneration"]
+	revision.Spec = originalrevision.Spec
+
+	return &revision
+}
+
+// Check if service exists
+func (cl *migrationClient) ServiceExists(name string) (bool, error) {
+	_, err := cl.client.Services(cl.namespace).Get(name, metav1.GetOptions{})
+	if api_errors.IsNotFound(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+// Get a config by service name
+func (cl *migrationClient) GetConfig(name string) (*serving_v1_api.Configuration, error) {
+	config, err := cl.client.Configurations(cl.namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return config, nil
+}
+
+// Get a service by name
+func (cl *migrationClient) GetService(name string) (*serving_v1_api.Service, error) {
+	service, err := cl.client.Services(cl.namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return service, nil
+}
+
+// Get service list
+func (cl *migrationClient) ListService() (*serving_v1_api.ServiceList, error) {
+	servicelist, err := cl.client.Services(cl.namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return servicelist, nil
+}
+
+// Create a service
+func (cl *migrationClient) CreateService(service *serving_v1_api.Service) (*serving_v1_api.Service, error) {
+	newserivce := cl.ConstructService(*service)
+	service, err := cl.client.Services(cl.namespace).Create(newserivce)
+	if err != nil {
+		return nil, err
+	}
+	return service, nil
+}
+
+// Delete a service by name
+func (cl *migrationClient) DeleteService(name string) error {
+	err := cl.client.Services(cl.namespace).Delete(name, &metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Get a revision by service name
+func (cl *migrationClient) GetRevision(name string) (*serving_v1_api.Revision, error) {
+	revision, err := cl.client.Revisions(cl.namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return revision, nil
+}
+
+// Create a revision
+func (cl *migrationClient) CreateRevision(revision *serving_v1_api.Revision, config_uuid types.UID) (*serving_v1_api.Revision, error) {
+	newrevision := cl.ConstructRevision(*revision, config_uuid)
+	revision, err := cl.client.Revisions(cl.namespace).Create(newrevision)
+	if err != nil {
+		return nil, err
+	}
+	return revision, nil
+}
+
+// Update the given revision
+func (cl *migrationClient) UpdateRevision(revision *serving_v1_api.Revision) error {
+	_, err := cl.client.Revisions(cl.namespace).Update(revision)
+	if err != nil {
+		return err
+	}
+	//return updateServingGvk(revision)
```

Comment:
What's this comment??

Reply:
fixed in the second commit: https://github.com/knative/client-contrib/pull/6/commits/e77b98b6cebdccc7a81fa81f9c508488611a2d3b